sympy/core/function.py

```diff
@@ -1855,10 +1855,28 @@ class Lambda(Expr):
     >>> f(*p)
     x + y*z
 
+    For having a function as its parameter:
+
+    >>> f = Lambda(lambda x, y: (x + y)**2)
+    >>> print(f)
+    Lambda((x, y), (x + y)**2)
+    >>> f(2,3)
+    25
+
     """
     is_Function = True
 
-    def __new__(cls, variables, expr):
+    def __new__(cls, *args):
+        if len(args) != 1:
+            variables, expr = args
+        else:
+            func = args[0]
+            if not callable(func):
```

Comment:
It needs to be a function (I think you can use `types.FunctionType`). Otherwise the `__code__` part won't work.